pkg/apis/monitoring/v1alpha1/scrapeconfig_types.go

```diff
@@ -906,3 +909,42 @@ type NomadSDConfig struct {
 	// +optional
 	EnableHTTP2 *bool `json:"enableHTTP2,omitempty"`
 }
+
+type LinodeSDConfig struct {
+	// BasicAuth information to use on every scrape request, required when role is robot.
+	// Role hcloud does not support basic auth.
+	// +optional
+	BasicAuth *v1.BasicAuth `json:"basicAuth,omitempty"`
+	// Authorization header configuration, required when role is hcloud.
+	// Role robot does not support bearer token authentication.
+	// +optional
+	Authorization *v1.SafeAuthorization `json:"authorization,omitempty"`
+	// Optional OAuth 2.0 configuration.
+	// Cannot be used at the same time as `basic_auth` or `authorization`.
+	// +optional
+	OAuth2 *v1.OAuth2 `json:"oauth2,omitempty"`
+	// +optional
+	Region *string `json:"region,omitempty"`
+	// ProxyConfig allows customizing the proxy behaviour for this scrape config.
+	// +optional
+	v1.ProxyConfig `json:",inline"`
+	// Configure whether HTTP requests follow HTTP 3xx redirects.
+	// +optional
+	FollowRedirects *bool `json:"followRedirects,omitempty"`
+	// TLS configuration applying to the target HTTP endpoint.
+	// +optional
+	TLSConfig *v1.SafeTLSConfig `json:"tlsConfig,omitempty"`
+	// The port number used if the query type is not SRV
+	// Ignored for SRV records
+	// +optional
+	Port *int `json:"port,omitempty"`
+	// The string by which Linode Instance tags are joined into the tag label
+	// +optional
```

Comment:
can you add a min length requirement

Reply:
The other instances of `TagSeparator` fields do not seem to have a validation for min length, should we do something about that? 

I see it is already `omitempty` anyway. Why is a min length required: as long as we set the min length to 1, isn't it as good as `omitempty`? Am I missing anything?

Reply:
> The other instances of TagSeparator fields do not seem to have a validation for min length, should we do something about that?

A thing to note for the consistency across SDs.

> Why is a min length required

the MinLength validation should only apply when the pointer isn't nil so you can either have the field undefined or defined but with at least 1 char.

Reply:
I am still finding it hard to digest, Is is possible to be for the field to be defined (non-nil) and yet not even 1 character long?